src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -178,4 +420,38 @@ public void Push(T item)
         item = Pop();
         return true;
     }
+
+    /// <summary>
+    ///  This is present to help the JIT with inlining methods that need to throw
+    ///  a <see cref="IndexOutOfRangeException"/>.
+    /// </summary>
+    [DoesNotReturn]
+    private static void ThrowIndexOutOfRangeException()
+        => throw new IndexOutOfRangeException();
+
+    [MemberNotNull(nameof(_builder))]
+    private void MoveInlineItemsToBuilder()
+    {
+        Debug.Assert(_builder is null);
+
+        var builder = ArrayBuilderPool<T>.Default.Get();
+
+        if (_capacity is int capacity)
+        {
+            builder.SetCapacityIfLarger(capacity);
+        }
+
+        // Add the inline items and clear their field values.
+        for (var i = 0; i < _inlineCount; i++)
+        {
+            builder.Add(this[i]);
```

Comment:
Super nit: Could call `GetInlineElement` here and avoid the checks in the indexer which we know already pass in this method.

However totally fair to leave this for future proofing (and i'm guessing probably a good chance the JIT elides the checks anyway, once it inlines the method)

Reply:
Good catch! I missed this when refactoring some code into `GetInlineElement`.